src/mongocxx/test/client_side_encryption.cpp

```diff
@@ -228,7 +228,7 @@ void run_datakey_and_double_encryption(Callable create_data_key,
     // AEAD_AES_256_CBC_HMAC_SHA_512-Deterministic, and the key_alt_name of provider_altname
     options::encrypt opts2{};
     opts2.algorithm(options::encrypt::encryption_algorithm::k_deterministic);
-    std::string altname{provider};
+    std::string altname{static_cast<std::string>(provider)};
```

Comment:
It looks like for C++17 and beyond, std::string provides a constructor that can take a string_view.  Here, though, I'm assuming we don't have that constructor available, and I'm not sure how this static_cast is legal without that constructor.

Let's do this instead:
`std::string altname{provider.data(), provider.size()};`

Reply:
Nice!